core/node/state_keeper/src/updates/l2_block_updates.rs

```diff
@@ -104,9 +106,17 @@ impl L2BlockUpdates {
         };
 
         let revert_reason = match &tx_execution_result.result {
-            ExecutionResult::Success { .. } => None,
-            ExecutionResult::Revert { output } => Some(output.to_string()),
-            ExecutionResult::Halt { reason } => Some(reason.to_string()),
+            ExecutionResult::Success { .. } => {
+                KEEPER_METRICS.inc_succeeded_txs();
+                None
+            }
+            ExecutionResult::Revert { output } => {
+                KEEPER_METRICS.inc_reverted_txs(vm_revert_reason_as_metric_label(output));
```

Comment:
Nit: It would help encapsulation to pass `&VmRevertReason` to `inc_reverted_txs()` and convert it to a label there; labels are responsibility of metrics.